src/concern/test.rs

```diff
@@ -210,3 +217,251 @@ async fn assert_event_contains_read_concern(client: &EventClient) {
+        UpdateOptions::builder()
+            .write_concern(
+                WriteConcern::builder()
+                    .w(Acknowledgment::Nodes(1))
+                    .journal(true)
+                    .build(),
+            )
+            .build(),
+    )
+    .await
+    .unwrap();
+    coll.update_one(
+        doc! { "foo": "baz" },
+        doc! { "$set": { "foo": "quux" } },
+        UpdateOptions::builder()
+            .write_concern(
+                WriteConcern::builder()
+                    .w(Acknowledgment::Nodes(1))
+                    .journal(false)
+                    .build(),
+            )
+            .build(),
+    )
+    .await
+    .unwrap();
+
+    assert_eq!(
+        buffer.write_concerns("update"),
+        vec![
+            doc! {
+                "w": 1,
+                "j": true,
+            },
+            doc! {
+                "w": 1,
+                "j": false,
+            },
+        ]
+    );
+}
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test)]
+#[cfg_attr(feature = "async-std-runtime", async_std::test)]
+#[function_name::named]
+async fn command_contains_write_concern_update_many() {
+    let _guard = LOCK.run_concurrently().await;
+    let (buffer, coll) = get_monitored_collection(function_name!()).await.unwrap();
+
+    coll.insert_many(&[doc! { "foo": "bar" }, doc! { "foo": "bar" }], None)
+        .await
+        .unwrap();
+    coll.update_many(
+        doc! { "foo": "bar" },
+        doc! { "$set": { "foo": "baz" } },
+        UpdateOptions::builder()
+            .write_concern(
+                WriteConcern::builder()
+                    .w(Acknowledgment::Nodes(1))
+                    .journal(true)
+                    .build(),
+            )
+            .build(),
+    )
+    .await
+    .unwrap();
+    coll.update_many(
+        doc! { "foo": "baz" },
+        doc! { "$set": { "foo": "quux" } },
+        UpdateOptions::builder()
+            .write_concern(
+                WriteConcern::builder()
+                    .w(Acknowledgment::Nodes(1))
+                    .journal(false)
+                    .build(),
+            )
+            .build(),
+    )
+    .await
+    .unwrap();
+
+    assert_eq!(
+        buffer.write_concerns("update"),
+        vec![
+            doc! {
+                "w": 1,
+                "j": true,
+            },
+            doc! {
+                "w": 1,
+                "j": false,
+            },
+        ]
+    );
+}
+
+async fn get_monitored_collection(name: &str) -> Result<(Arc<CommandBuffer>, Collection)> {
+    let buffer = Arc::new(CommandBuffer::new());
+    let mut options = CLIENT_OPTIONS.clone();
+    options.command_event_handler = Some(buffer.clone());
+    let client = TestClient::with_options(Some(options)).await;
+    let db = client.database("test");
+    let coll: Collection = db.collection(name);
+    coll.drop(None).await?;
+    Ok((buffer, coll))
+}
+
+struct CommandBuffer {
```

Comment:
We actually already have some helper types to facilitate this kind of thing, namely `EventClient` (which is a `Client` wrapper which automatically records all events) and `EventHandler` (which is just a handler that can be used when more granularity over the recorded events is required). In this case, I think `EventClient` should suffice. Check out [`index_options_defaults_test`](https://github.com/mongodb/mongo-rust-driver/blob/master/src/test/db.rs#L368) for an example of how to use it.

Reply:
Handy, thank you!  Updated.